File: app/src/main/java/medlife/com/br/adapter/CategoryAdapter.java
package medlife.com.br.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;
import medlife.com.br.R;
import medlife.com.br.model.Category;

public class CategoryAdapter extends RecyclerView.Adapter<CategoryAdapter.CategoryViewHolder> {
    private Context context;
    private List<Category> categoryList;

    public CategoryAdapter(Context context, List<Category> categoryList) {
        this.context = context;
        this.categoryList = categoryList;
    }

    @NonNull
    @Override
    public CategoryViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_subcategory, parent, false);
        return new CategoryViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull CategoryViewHolder holder, int position) {
        Category category = categoryList.get(position);
        holder.imageSubcategory.setImageResource(category.getImageResId());
        holder.textSubcategory.setText(category.getName());
    }

    @Override
    public int getItemCount() {
        return categoryList.size();
    }

    public static class CategoryViewHolder extends RecyclerView.ViewHolder {
        ImageView imageSubcategory;
        TextView textSubcategory;

        public CategoryViewHolder(@NonNull View itemView) {
            super(itemView);
            imageSubcategory = itemView.findViewById(R.id.imageSubcategory);
            textSubcategory = itemView.findViewById(R.id.textSubcategory);
        }
    }
} 

File: app/src/main/java/medlife/com/br/adapter/CategoryAdapter.kt
package medlife.com.br.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import medlife.com.br.R
import medlife.com.br.model.Category

class CategoryAdapter(
    private val context: Context?,
    private val categoryList: List<Category>
) : RecyclerView.Adapter<CategoryAdapter.CategoryViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CategoryViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.item_subcategory, parent, false)
        return CategoryViewHolder(view)
    }

    override fun onBindViewHolder(holder: CategoryViewHolder, position: Int) {
        val category = categoryList[position]
        holder.imageSubcategory.setImageResource(category.imageResId)
        holder.textSubcategory.text = category.name
    }

    override fun getItemCount(): Int {
        return categoryList.size
    }

    class CategoryViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val imageSubcategory: ImageView = itemView.findViewById(R.id.imageSubcategory)
        val textSubcategory: TextView = itemView.findViewById(R.id.textSubcategory)
    }
}
